### ISSUE TITLE
Feedback

### ISSUE THREAD

**Issue by chyt** [OpenLiberty/draft-guide-arquillian-managed#20]
- What you’ll learn
    - Link to https://github.com/OpenLiberty/liberty-arquillian/tree/master/liberty-managed
    - “features in the Liberty plugin that simplifies the process” -> “features in the Liberty Maven Plugin that simplify the process”
    - Second paragraph - need to clarify that Arquillian is a JUnit runner and the test components are actually JUnit components, not Arquillian components. Maybe should also mention that there is a TestNG Arquillian runner as well
    - Second paragraph grammar - “Arquillian sets up the test environment and able to handle the application” -> “Arquillian sets up the test environment and is able to handle the application”
    - Third paragraph - don’t think mentioning “DeployableContainer” is necessary. Just explain that Arquillian allows different app servers to implement their own embedded, managed, or remote containers, and that in this guide we will be explaining how to use the managed Liberty container
EC: Just thought it's good to explain what Liberty Arquillian Managed container is. I modified the sentence, see what do you think. 
- Setting up your Arquillian Liberty Managed container -> too wordy, consider renaming to something like “Configuring Arquillian with Liberty”
    - “You will need to configure your build tool” -> why not just say “configure Maven”
    - Spelling error - “Arquillina”
    - “You will start configuring your test build with the Maven build tool” -> just say “configuring your test build with Maven”.
    - Arquillian BOM section - “defines the versions of dependencies” -> “defines the versions of Arquillian dependencies”
- Developing Arquillian Tests
    - In the deployment, simplify by just specifying packages instead of each class individually
    - “Notice the arquillian-managed.war WAR name in the web archive, this is necessary, if you don’t want a randomly generated web archive name.” - split into two sentences -> “…in the web archive. This is necessary…”
    - “If you want to point to a file system to include the classes for your test, you can follow the Arquillian guide to do that.” - don’t think it’s necessary to say this. I would just say at the end of the paragraph to refer to the Arquillian Shrinkwrap guide for information on more complex archive configuration.
    - “The @RunAsClient annotation added in this test case is to indicate that this test case is to be run as a client” - using “run as a client” to explain “@RunAsClient” is not a very good explanation. Instead say that the test will be run on the client side, against the container. 
    - “http://localhost:9080/<WARNAME>/inventory/systems/{hostname}” and the other endpoint shows up as a broken link. Is this intentional?
EC: This link doesn't meant to be clickable, because the {hostname} is not filled. Also the service wouldn't be available unless it's during the tests are being run. 
    - “The @InSequence annotation from Arquillian’s JUnit integration guarantees the test sequence” - what does “Arquillian’s JUnit integration” mean? Don’t think it is necessary to say that, just say “The @InSequence annotation guarantees the test sequence”
EC: the term "Arqullian's JUnit integration" was from the Arquillian guide, http://arquillian.org/guides/getting_started_rinse_and_repeat/. I think it makes it more specific that this annotation is from Arquillian and that provides the test sequence guarentee, cus JUnit tests don't have this. 
- Running the tests
    - Capitalize “maven”
    - “In the tests output, you see the application” - should probably be “in the test output”
    - “the test application is automatically destroyed” - should be “undeployed” not “destroyed”
    - “If you make changes to the test cases, you can verify that the tests pass using the Maven verify goal” - this is not correct. “verify” is a phase not a goal, and it will call “install” which is actually doing the test case execution. Why not just tell them to run “mvn clean install” again?

**Comment by evelinec** [comment]
Addressed everything except for this one: 
- In the deployment, simplify by just specifying packages instead of each class individually

I think it's clearer to list all the classes as an intro guide. 

PR will be submitted soon.

**Comment by evelinec** [comment]
@chyt can you review the fixes for this issue and let me know if there's anything? Thanks.

**Comment by chyt** [comment]
- What you’ll learn
    - “Arquillian tests are very similar to the common unit tests, however, Arquillian sets up the test environment and is able to handle the application server life cycle for you, so you can focus on writing tests.” - This is a run-on sentence, should be broken up into two sentences. Also not sure if saying that Arquillian is similar to “common unit tests” since you can use Arquillian for both unit and integration tests. Should emphasize that the focus of Arquillian is running in-container tests. 
    - “The Arquillian JUnit test has three components” - not completely true since there are other components. Maybe just say that “An Arquillian JUnit test class has three core components” instead
    - “which is an adaptor to run Arquillian tests with Liberty.” - maybe be more specific and say “run Arquillian tests on or against a locally managed Liberty runtime.” 
    - Make sure that the naming of the container is consistent. In the last paragraph you say “Liberty Arquillian Managed” but previously you say “Arquillian Liberty Managed”. I think the container should be consistently referred to as the “Arquillian Liberty Managed” container
- Configuring Arquillian with Liberty
    - I do not believe using “<version>RELEASE</version>” for a parent dependency is valid - see https://stackoverflow.com/a/1172371/634324
    - Update “arquillian-liberty-managed-junit” dependency to the latest version (1.1.3). Might also want to mention that this corresponds to version 1.0.3 of the container
    - “Finally, the maven-failsafe-plugin is used to run your JUnit integration tests” - this might be a bit ambiguous. Maybe be more clear and say “run your Arquillian integration tests using JUnit”. 
    - “The test server configuration is disable” -> disabled not disable
- Developing Arquillian Tests
    - “The @RunAsClient annotation added in this test case is to indicate” -> just say “indicates” instead of “is to indicate”
    - “which is a functional test case to test the functions in the InventoryResource class.” - this is redundant, just say “which is a functional test case for the InventoryResource class”
    - “The functional test case does not do the action to store any system in the inventory, the one system that’s found was from the endpoint test case that was run prior to this test case. “ - this sentence is wordy and I’m not sure if it’s necessary. Why does this need to be said? You’ve already described what the test case does in the previous sentence. 
    - “The @InSequence annotation from Arquillian’s JUnit integration guarantees the test sequence. “ - just say “The @InSequence Arquillian annotation guarantees the test sequence”.
    - Remove/change the line about mvn verify, as I described in my previous feedback #20

**Comment by evelinec** [comment]
Point not handled yet: 
- [ ] I do not believe using “RELEASE” for a parent dependency is valid - see https://stackoverflow.com/a/1172371/634324
EC: The team will determine and handle this all together in multipane.

**Comment by evelinec** [comment]
Opened #25 for the one point that didn't addressed here. Close this issue.
